### PR TITLE
Implement liveness-check

### PR DESCRIPTION
**Pull request by jsantos17**
[ch798]

### CODE REVIEW

File: datasource/src/main/scala/quasar/physical/s3/S3DataSourceModule.scala [hunk 1]
```diff
@@ -43,13 +48,21 @@ object S3DataSourceModule extends LightweightDatasourceModule {
       : F[InitializationError[Json] \/ Disposable[F, Datasource[F, Stream[F, ?], ResourcePath, Stream[F, Data]]]] = {
     config.as[S3Config].result match {
       case Right(s3Config) => {
-        Http1Client[F]() map { client =>
-          val ds: Datasource[F, Stream[F, ?], ResourcePath, Stream[F, Data]] =
-            new S3DataSource[F](client, s3Config)
+        Http1Client[F]() flatMap { client =>
+          val s3Ds = new S3DataSource[F](client, s3Config)
+          val ds: Datasource[F, Stream[F, ?], ResourcePath, Stream[F, Data]] = s3Ds
```
Comment (rintcius) [inline datasource/src/main/scala/quasar/physical/s3/S3DataSourceModule.scala]: I think you can merge `s3Ds` and `ds` together into 1 val?

File: datasource/src/test/scala/quasar/physical/s3/S3DataSourceModuleSpec.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+import argonaut.Json
+import cats.effect.IO
+import org.specs2.mutable.Specification
+import shims._
+
+class S3DataSourceModuleSpec extends Specification {
+  import S3DataSourceModuleSpec._
+
+  "validates credentials when creating a new S3DataSource" >> {
```
Comment (rintcius) [inline datasource/src/test/scala/quasar/physical/s3/S3DataSourceModuleSpec.scala]: maybe add a few other test cases too? (like testing `"bucket" -> Json.jString("no url")` which is also mentioned in the ticket and a valid url that's not a bucket, like `"bucket" -> Json.jString("https://google.com")`)

File: datasource/src/main/scala/quasar/physical/s3/S3DataSourceModule.scala (hunk 1 above)
Comment (jsantos17) [reply datasource/src/main/scala/quasar/physical/s3/S3DataSourceModule.scala]: I think we need this so we can upcast `S3DataSource[F]` to `Datasource[F, Stream[F, ?], ResourcePath, Stream[F, Data]]`?

**Comment by jsantos17** [comment]
@rintcius added a new test to check non-bucket URIs. With regard to `ds` and `s3Ds`, I think I need values with both types? I need to return something of type `Datasource[F, Stream[F, ?], ResourcePath, Stream[F, Data]]` and I need something of type `S3DataSource[F]` to call `isLive` on.

Can I upcast `S3DataSource[F]` in a different way?

File: datasource/src/main/scala/quasar/physical/s3/S3DataSourceModule.scala (hunk 1 above)
Comment (rintcius) [reply datasource/src/main/scala/quasar/physical/s3/S3DataSourceModule.scala]: Ah right, just adding a type annotation for `s3Ds` would have made that difference more explicit.

Another option, without using `ds`, would be:

```
s3Ds.isLive.ifM({
            val disposable = Disposable((s3Ds: Datasource[F, Stream[F, ?], ResourcePath, Stream[F, Data]]), client.shutdown)
```

But fine with me to leave like this.

**Pull request merged**